komodo/cli.py

```diff
@@ -62,190 +103,308 @@ def _print_timings(
+    download_mode: bool, build_mode: bool
+) -> None:
+    if download_mode and not build_mode:
+        sys.exit(0)
+
+
+@profile_time("Fetching all packages")
+def download_packages(
+    release_file_content: Mapping[str, str],
+    repository_file_content: Mapping[str, Mapping[str, Union[str, Sequence[str]]]],
+    download_destination: str,
+    pip_executable: str = "pip",
+) -> Dict[str, str]:
+    """Downloads all PyPI packages to destination. Tries to download other
+        packages to destination too.
+    Git packages are collected, and a dict of all git hashes is returned.
+
+    Args:
+    ----
+    release_file_content: A mapping of all packages and versions to be
+        included in the release.
+    repository_file_content: A mapping of all available packages and
+        versions that can be included in a release.
+    download_destination: A string of the path where packages should
+        be downloaded to
+    pip_executable: A string of the pip executable to use. Defaults to 'pip'
+
+    Returns:
+    --
+    Dict of git hashes found
+    """
+    git_hashes = fetch(
+        release_file_content,
+        repository_file_content,
+        outdir=download_destination,
+        pip=pip_executable,
+    )
+
+    return git_hashes
+
+
+@profile_time("Building non-pip part of komodo in workspace")
+def _make(*args, **kwargs) -> None:
+    make(*args, **kwargs)
+
+
+def build_packages_and_move_to_release_path(
+    args: KomodoNamespace, data: Data, tmp_prefix: Path
+):
     fakeroot = Path(args.release).resolve()
-    if args.build or not args.install:
-        start_time = datetime.datetime.now()
-        make(
-            args.pkgs.content,
-            args.repo.content,
-            data,
-            prefix=str(tmp_prefix),
-            dlprefix=args.downloads,
-            builddir=args.tmp,
-            jobs=args.jobs,
-            cmk=args.cmake,
-            pip=args.pip,
-            fakeroot=str(fakeroot),
-        )
-        timings.append(
-            (
-                "Building non-pip part of komodo in workspace",
-                datetime.datetime.now() - start_time,
-            ),
-        )
-        _print_timings(timings[-1])
+    _make(
+        args.pkgs.content,
+        args.repo.content,
+        data,
+        prefix=str(tmp_prefix),
+        dlprefix=args.downloads,
+        builddir=args.tmp,
+        jobs=args.jobs,
+        cmk=args.cmake,
+        pip=args.pip,
+        fakeroot=str(fakeroot),
+    )
 
-        shell(f"mv {args.release + str(tmp_prefix)} {args.release}")
-        with contextlib.suppress(OSError):
-            os.removedirs(f"{args.release + str(tmp_prefix.parent)}")
+    shell(f"mv {args.release + str(tmp_prefix)} {args.release}")
+    with contextlib.suppress(OSError):
+        os.removedirs(f"{args.release + str(tmp_prefix.parent)}")
 
-    if args.build and not args.install:
+
+def exit_on_build_only_and_no_install_run(build_mode: bool, install_mode: bool) -> None:
```

Comment:
Does this need to be a stand alone function?

Reply:
I found it easier to read, but that might only be me. Do you think I should revert it? 
I was also considering putting this out in a separate function ```if args.download or (not args.build and not args.install):```. Do you think that would be a good idea?

Reply:
I would maybe call that function 
```
def download_mode_or_unspecified_mode(args) -> bool:
    return args.download or (not args.build and not args.install)
```

and change the main function:
```
if download_mode_or_unspecified_mode(args):
    git_hashes = download_packages(
            args.pkgs.content,
            args.repo.content,
            download_destination=args.downloads,
            pip_executable=args.pip,
        )
    exit_on_download_only_and_no_build_run(args.download, args.build)
```

Reply:
> Do you think I should revert it?

Let's keep it then as I thought it makes the readability low, but the function name is quite self explanatory.

> I was also considering putting this out in a separate function if args.download or (not args.build and not args.install):. Do you think that would be a good idea?

Maybe it would, but the those `helping` functions should have the same semantics. Eg. 
`if is_download_mode(args)` -> 
and the same would go for the one above (?):
`if is_build_only(args): exit(0)` 

but not sure

Reply:
Changed it now. What do you think?